### ISSUE TITLE
Pasting deletes text

### ISSUE THREAD

**Issue by ygoe** [microsoft/vscode#155660]
Issue Type: <b>Bug</b>

Prerequisites: Enable format on paste and on type for JavaScript files.

1. In a JavaScript file, enter: `match(/x/);`
2. Copy the "x" with Ctrl+C
3. After the "x", paste with Ctrl+V

The entire regular expression now disappears. Also happens with much longer lines and regexps. A similar issue also exists in JavaScript sections of .cshtml files in Visual Studio 2022, where pasting also deletes parts of the code. Is there a pattern behind that? Microsoft products now eating my code. All of them!

VS Code version: Code 1.69.2 (3b889b090b5ad5793f524b5d1d39fda662b96a2a, 2022-07-18T16:12:52.460Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-3770 CPU @ 3.40GHz (8 x 3400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.71GB (5.85GB free)|
|Process Argv|--crash-reporter-id 1b00a758-578c-4b77-a258-fe145c18e004|
|Screen Reader|no|
|VM|29%|
</details><details><summary>Extensions (22)</summary>

Extension|Author (truncated)|Version
---|---|---
esp-idf-extension|esp|1.4.0
vscode-todo-highlight|jgc|2.0.4
cmake-language-support-vscode|jos|0.0.4
ecdc|mit|1.8.0
vscode-language-pack-de|MS-|1.69.7130931
csharp|ms-|1.25.0
vscode-dotnet-runtime|ms-|1.5.0
python|ms-|2022.10.1
vscode-pylance|ms-|2022.7.30
jupyter|ms-|2022.6.1201981810
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.8
remote-containers|ms-|0.241.3
remote-ssh|ms-|0.84.0
remote-ssh-edit|ms-|0.80.0
remote-wsl|ms-|0.66.3
vscode-remote-extensionpack|ms-|0.21.0
cmake-tools|ms-|1.11.26
cpptools|ms-|1.10.8
better-pageup-pagedown|rus|0.0.1
jinjahtml|sam|0.17.0
cmake|twx|0.0.17

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscorecescf:30445987
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30449410
vscscmwlcmt:30465135
cppdebug:30492333
pylanb8912:30529769
vsclangdf:30486550

```

</details>

**Comment by mjbvz** [comment]
Please try upgrading your workspace to use `typescript@next` by installing [this extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next). Does that fix the issue?

**Comment by ygoe** [comment]
Yes, still happens.

**Comment by mjbvz** [comment]
Duplicate of https://github.com/microsoft/TypeScript/issues/50076

Should be fixed with more recent TS nightly builds

**Comment by freaker2k7** [comment]
Still happens on latest version on MAC 😭

**Comment by ygoe** [comment]
Fixed in VSCode 1.71.0
